src/Sylius/Bundle/SettingsBundle/Manager/SettingsManager.php

```diff
@@ -101,20 +102,30 @@ public function __construct(
     /**
      * {@inheritdoc}
      */
-    public function loadSettings($namespace, $ignoreUnknown = true)
+    public function load($schemaAlias, $namespace = null, $ignoreUnknown = true)
     {
-        if (isset($this->resolvedSettings[$namespace])) {
-            return $this->resolvedSettings[$namespace];
+        $schema = $this->schemaRegistry->get($schemaAlias);
+
+        $resolver = $this->defaultResolver;
+
+        if ($this->resolverRegistry->has($schemaAlias)) {
+            $resolver = $this->resolverRegistry->get($schemaAlias);
         }
 
-        if ($this->cache->contains($namespace)) {
-            $parameters = $this->cache->fetch($namespace);
-        } else {
-            $parameters = $this->getParameters($namespace);
-            $this->cache->save($namespace, $parameters);
+        // try to resolve settings for schema alias and namespace
+        $settings = $resolver->resolve($schemaAlias, $namespace);
+
+        if (!$settings) {
+            $settings = $this->settingsFactory->createNew();
+            $settings->setSchemaAlias($schemaAlias);
         }
 
-        $schema = $this->schemaRegistry->getSchema($namespace);
+        // map parameters to a plain php array
+        $parameters = $settings->getParameters()
```

Comment:
Yes it is indeed too much for something so simple. If we want to make it simpeler we would need, as you were saying, remove the parameter model all together and maybe make is a serialized array inside the settings model. While this is faster and might make it simpelere you cannot easily extend a single parameter. Right now you can override the `Parameter` model and do specific stuff there. 

It really depends what you would prefer. If extension of the parameter model is not required, then we can get rid of the parameter factory, it's model etc.